addons/website/models/ir_ui_view.py

```diff
@@ -45,6 +45,30 @@ def _compute_first_page_id(self):
         for view in self:
             view.first_page_id = self.env['website.page'].search([('view_id', '=', view.id)], limit=1)
 
+    def create(self, vals_list):
+        """
+        SOC for ir.ui.view creation. If a view is created without a website_id,
+        it should get one if one is present in the context. Also check that
+        an explicit website_id in create values matches the one in the context.
+        """
+        website_id = self.env.context.get('website_id', False)
+        if not website_id:
+            return super().create(vals_list)
+
+        if 'website_id' not in vals_list:
+            # Automatic addition of website ID during view creation if not
+            # specified but present in the context
+            vals_list['website_id'] = website_id
+        else:
+            # If website ID specified, automatic check that it is the same as
+            # the one in the context. Otherwise raise an error.
+            new_website_id = vals_list['website_id']
+            if not new_website_id:
+                raise ValidationError("Trying to create a generic view from a website %s environment" % (website_id))
```

Comment:
Isnt this missing translation ?

Reply:
Mmh, did not occur to me this could be shown to users but I guess it's possible. But for me it's like a traceback, we don't translate "Variable xy is undefined"... I can add the `_(` and be done with this but should we really add this to be translated?

Reply:
Any ValidationError or UserError is supposed to be translated AFAIK

Reply:
Maybe it's not `ValidationError` I should use then, but I don't see any more suitable. Any error class so that the code crashes but the error is not displayed to the user?

Reply:
Why would we want to hide the error to the user? He would just be lost forever with no indication no?

Reply:
A framework should never translate its errors IMO... this is not a warning message for the user or an "error" he can fix himself, this is a *code* error that will block him no matter what. This is the same as making a type in your code. If an user reports that error to the support, I don't expect to try and find the inverse translation to find the error in the code :/ If the code is properly done, this error should never be thrown, hence there is no reason to translate that text.

Reply:
Nice catch, `ValidationError`s are indeed displayed like a user warning. I am changing to `ValueError` which seems more appropriate and displays the error as a traceback if it ever occurs :+1: